src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Dec2Hex.txt

```diff
@@ -0,0 +1,184 @@
+﻿﻿// ******** INTEGER PARAMETERS WITHOUT PLACES ********
+
+>> Dec2Hex(0)
+"0"
+
+>> Dec2Hex(Blank())
+"0"
+
+// Positive Integer Parameters
+>> Dec2Hex(165)
+"A5"
+
+>> Dec2Hex(3213)
+"C8D"
+
+>> Dec2Hex(549755813887)
+"7FFFFFFFFF"
+
+// Negative Integer Parameters
+>> Dec2Hex(-1)
+"FFFFFFFFFF"
+
+>> Dec2Hex(-165)
+"FFFFFFFF5B"
+
+>> Dec2Hex(-3213)
+"FFFFFFF373"
+
+>> Dec2Hex(-549755813888)
+"8000000000"
+
+// ******** DECIMAL PARAMETERS ********
+
+>> Dec2Hex(3.74)
+"3"
+
+>> Dec2Hex(-3.74)
+"FFFFFFFFFC"
+
+
+// ******** INTEGER PARAMETERS WITH PLACES ********
+
+>> Dec2Hex(45780)
+"B2D4"
+
+>> Dec2Hex(45780, 0)
+"B2D4"
+
+>> Dec2Hex(45780, 3)
+#Error(Kind=InvalidArgument)
+
+>> Dec2Hex(45780, 4)
+"B2D4"
+
+>> Dec2Hex(45780, 5)
+"0B2D4"
+
+>> Dec2Hex(45780, 6)
+"00B2D4"
+
+>> Dec2Hex(45780, 10)
+"000000B2D4"
+
+>> Dec2Hex(45780, -6)
+#Error(Kind=Numeric)
+
+>> Dec2Hex(45780, 11)
+#Error(Kind=Numeric)
+
+// Places are ignored when number is negative
+>> Dec2Hex(-45780)
+"FFFFFF4D2C"
+
+>> Dec2Hex(-45780, 4)
+"FFFFFF4D2C"
+
+>> Dec2Hex(-45780, 5)
+"FFFFFF4D2C"
+
+>> Dec2Hex(-45780, 6)
+"FFFFFF4D2C"
+
+>> Dec2Hex(-45780, 11)
+#Error(Kind=Numeric)
+
+// ******** NULL and ERROR PARAMETERS ********
+>> Dec2Hex(1/0)
+#Error(Kind=Div0)
+
+>> Dec2Hex(-549755813889)
+#Error(Kind=Numeric)
+
+>> Dec2Hex(549755813888)
+#Error(Kind=Numeric)
+
+>> Dec2Hex(If(1/0<2,["true"]))
+#Error(Kind=Div0)
+
+>> Dec2Hex(If(1<0,["true"]))
+Blank()
+
+>> Dec2Hex(Filter(["true","false"], Len(Value) > 10))
+[]
+
+>> Dec2Hex(If(1/0<2,[1]))
+#Error(Kind=Div0)
+
+>> Dec2Hex(If(1<0,[1]))
+Blank()
+
+>> Dec2Hex(Filter([1,2,3], Value > 10))
+[]
+
+>> Dec2Hex(If(1/0<2,[12, 56]), 1)
+#Error(Kind=Div0)
+
+>> Dec2Hex(1234, If(Sqrt(-1)<0,[1, 2, 3]))
+#Error(Kind=Numeric)
+
+>> Dec2Hex([12, 56], 1/0)
+[Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue]
+
+>> Dec2Hex(Sqrt(-1), [1, 2, 3])
+[Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue,Microsoft.PowerFx.Types.ErrorValue]
+
+>> Dec2Hex([123, 234], [3, 2, 3])
+["07B","EA",Microsoft.PowerFx.Types.ErrorValue]
+
+
+// ******** STRING PARAMETERS ********
+
+>> Dec2Hex("sampleText")
+#Error(Kind=InvalidArgument)
+
+>> Dec2Hex("4")
+"4"
+
+>> Dec2Hex("14", 4)
+"000E"
+
+>> Dec2Hex("15", "6")
+"00000F"
+
+>> Dec2Hex("-9")
+"FFFFFFFFF7"
+
+// ******** BOOLEAN PARAMETERS ********
+
+>> Dec2Hex(true)
+"1"
+
+>> Dec2Hex(false)
+"0"
+
+>> Dec2Hex("true")
+#Error(Kind=InvalidArgument)
+
+>> Dec2Hex("false")
+#Error(Kind=InvalidArgument)
+
+// ******** FUNCTION VALUE AS PARAMETERS ********
+
+>> Dec2Hex(Ln(2))
+"0"
+
+>> Dec2Hex(Power(2,2))
+"4"
+
+// ******** TABULAR VALUE AS PARAMETERS ********
+
+>> Dec2Hex([10,255,-16])
+["A","FF","FFFFFFFFF0"]
+
+>> Dec2Hex([1,1e-45,1e45,2])
```

Comment:
New test to validate that we work with different input schema (as long as it has a single numeric column):

    >> Dec2Hex(Table({a:10},{a:255},{a:-16}))
    ["A","FF","FFFFFFFFF0"]

    >> Dec2Hex(Table({a:10},{a:11},{a:12}), Table({b:2},{b:5},{b:10))
["0A","0000B","000000000C"]